src/wireviz/svgembed.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import re
+import base64
+from pathlib import Path
+
+mime_subtype_replacements = {'jpg': 'jpeg', 'tif': 'tiff'}
+
+def embed_svg_images(svg_in: str, base_path: Path):
+    # first, find any image references in SVG data, and cache the respective base64-encoded image
+    images_b64 = {}  # cache of base64-encoded images
+    re_xlink=re.compile(r"xlink:href=\"(?P<URL>.*?)\"", re.IGNORECASE)
+    for xlink in re_xlink.finditer(svg_in):
+        imgurl = xlink.group('URL')
+        if not imgurl in images_b64:  # only encode/cache every unique URL once
+            if not Path(imgurl).is_absolute():
+                imgurl_abs = (Path(base_path) / imgurl).resolve()
+            else:
+                imgurl_abs = Path(imgurl)
+
+            with open(imgurl_abs,'rb') as img:
+                img_bin = img.read()
+                img_b64 = base64.b64encode(img_bin)
+                img_str = img_b64.decode('utf-8')
+            images_b64[imgurl] = img_str
+    # second, replace links with the base64-encoded data
+    svg_out = svg_in
+    for url, b64 in images_b64.items():
+        svg_out = svg_out.replace(url, f'data:image/{get_mime_subtype(url)};base64, {b64}')
+
+    return svg_out
+
+
+def get_mime_subtype(filename: Path):
+    mime_subtype = str(Path(filename).suffix[1:]).lower()  # remove `.`
+    if mime_subtype in mime_subtype_replacements:
+        mime_subtype = mime_subtype_replacements[mime_subtype]
```

Comment:
This could have been simplified to
```python
mime_subtype = mime_subtype_replacements.get(mime_subtype, mime_subtype)
```
but I decided against it for readability.

Reply:
Thanks. I'll leave it as it is now :)